### PR TITLE
[cmake] mark dirty tree in the version string

### PR DESCRIPTION
**Pull request by Saviq**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -47,7 +47,7 @@ add_subdirectory(3rd-party)
 find_package(Qt5Core REQUIRED)
 find_package(Qt5Network REQUIRED)
 
-execute_process(COMMAND git describe
+execute_process(COMMAND git describe --dirty
```
Comment (ricab) [inline CMakeLists.txt]: Won't this be wrong when the version is not dirty (i.e. an exact tag)?

**Comment by ricab** [comment]
For my understanding, what was the motivation for this?

File: CMakeLists.txt (hunk 1 above)
Comment (Saviq) [reply CMakeLists.txt]: No, see [`man git-describe`](https://git-scm.com/docs/git-describe#git-describe---dirtyltmarkgt)

Comment (ricab) [reply CMakeLists.txt]: Hmm, my earlier comment seems to have been lost. Anyway, I just said you were right, that I had seen the man but had overlooked that bit.

**Pull request merged**